src/renderer/atlas/AtlasEngine.r.cpp

```diff
@@ -243,7 +243,18 @@ void AtlasEngine::_renderWithCustomShader() const
         // PS: Pixel Shader
         _r.deviceContext->PSSetShader(_r.customPixelShader.get(), nullptr, 0);
         _r.deviceContext->PSSetConstantBuffers(0, 1, _r.customShaderConstantBuffer.addressof());
-        _r.deviceContext->PSSetShaderResources(0, 1, _r.customOffscreenTextureView.addressof());
+        if (_r.customShaderTexture.TextureView)
+        {
+            ID3D11ShaderResourceView* const resourceViews[]{
+                _r.customOffscreenTextureView.get(),
+                _r.customShaderTexture.TextureView.get(),
+            };
+            _r.deviceContext->PSSetShaderResources(0, 2, static_cast<ID3D11ShaderResourceView* const*>(resourceViews));

```

Comment:
You could deduplicate this code by making the length dependent on the existence of the texture view like so:
```cpp
ID3D11ShaderResourceView* const views[]{
    _r.customOffscreenTextureView.get(),
    _r.customShaderTexture.TextureView.get(),
};
const UINT numViews = _r.customShaderTexture.TextureView ? 2 : 1;
_r.deviceContext->PSSetShaderResources(0, numViews, &views[0]);
```

Reply:
Done